### PR TITLE
Review messaging in KN-PMP

### LINKED ISSUE #425: KN-PMP – review warnings & error messages
We should review the [KN-PMP](https://github.com/pysal/spopt/blob/bf7134a33794e9a02377e95563faaa48d74a486d/spopt/locate/p_median.py#L546) warnings & error messages for grammar, formatting, and proper variable/parameters naming.

xref #414

### PR DESCRIPTION
**Pull request by jGaboardi**
This MR:
* resolves #425 
* does some more docstring clean up

xref #414

### CODE REVIEW

**Comment by jGaboardi** [comment]
I need to update the test for this. Will push up a fix shortly.

**Review by ljwolf — APPROVED** [review]

File: spopt/locate/p_median.py [hunk 1]
```diff
@@ -547,7 +547,7 @@ def solve(self, solver: pulp.LpSolver, results: bool = True):
 class KNearestPMedian(PMedian):
     r"""
     Implement the P-Median Model with Near-Far Cost Allocation and solve it.
-    The model is adapted from :cite:`richard_2018`, can be formulated as:
+    The model is adapted from :cite:`richard_2018`, and can be formulated as:
```
Comment (ljwolf) [inline spopt/locate/p_median.py:550]: This should probably be keyed "church_2018", no?

Comment (ljwolf) [reply spopt/locate/p_median.py:550]: Anyway, I don't mind if it's not.

Comment (jGaboardi) [reply spopt/locate/p_median.py:550]: Looks like it snuck in as [`richard_2018 `](https://github.com/pysal/spopt/blob/a330389c767a883600af8cb8b76fca8d63610c3f/docs/_static/references.bib#L171).

Comment (jGaboardi) [reply spopt/locate/p_median.py:550]: I'll throw in a quick PR for a fix.

**Pull request merged**